x/group/server/proposal_executor.go

```diff
@@ -0,0 +1,42 @@
+package server
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/regen-network/regen-ledger/x/group"
+)
+
+// ensureMsgAuthZ checks that if a message requires signers that all of them are equal to the given group account.
+func ensureMsgAuthZ(msgs []sdk.Msg, groupAccount sdk.AccAddress) error {
+	for i := range msgs {
+		for _, acct := range msgs[i].GetSigners() {
+			if !groupAccount.Equals(acct) {
+				return errors.Wrap(errors.ErrUnauthorized, "msg does not have group account authorization")
+			}
+		}
+	}
+	return nil
+}
+
+// DoExecuteMsgs routes the messages to the registered handlers. Messages are limited to those that require no authZ or
+// by the group account only. Otherwise this gives access to other peoples accounts as the sdk ant handler is bypassed
+func DoExecuteMsgs(ctx sdk.Context, router sdk.Router, groupAccount sdk.AccAddress, msgs []sdk.Msg) ([]sdk.Result, error) {
+	results := make([]sdk.Result, len(msgs))
+	if err := ensureMsgAuthZ(msgs, groupAccount); err != nil {
+		return nil, err
+	}
+	for i, msg := range msgs {
+		handler := router.Route(ctx, msg.Route())
```

Comment:
should we be using ADR 031 here?

Reply:
Something similar to what's in `BaseApp.runMsgs` instead? I remember we had previous discussions to keep using `sdk.Router` for now but happy to upgrade that if that makes sense.

Reply:
I think it's fine for now, but can you add a follow up issue to address later?

Reply:
alright, added: https://github.com/regen-network/regen-ledger/issues/225